### ISSUE TITLE
WP configuration

### ISSUE THREAD

**Issue by Able2Code** [openbridge/nginx#36]
By installing Wordpress and using ob_php-fpm container, i'm getting the mixed-content error over the entire site. How to correct this error?
Thank you in advance !

**Comment by Able2Code** [comment]
Solved by editing wp-config.php 

define('FORCE_SSL_ADMIN', true);
if ($_SERVER['HTTP_X_FORWARDED_PROTO'] == 'https') $_SERVER['HTTPS']='on';

**Comment by Able2Code** [comment]
Hi @tspicer , how to prevent nginx to cache wordpress pages when i'm logged as administrator?
i'm using the php configuration with redis, in many cases when i visit any site page, the cache save also the wp admin toolbar header , by visiting in incognito i can see the cached wp admin toolbar.
I see also you wordpress dedicated git, this config will fix my problem? can i use you installer in any vps ?

**Comment by wualex** [comment]
@Able2Code how did you solve the 502 issue? have the same problem

**Comment by tspicer** [comment]
Since this is an NGINX (not Wordpress) repo, I will do my best to assist.

It is hard to troubleshoot these errors given the variations of configs. I'm happy to help, but please post configs for NGINX and wordpress. For example, in NGINX you can set it to not cache anything /wp-admin*. Also, if you have a WP installer you are using, please share that as well.

**Comment by wualex** [comment]
This would be my docker compose file

`version: '3'

services:
  web:
    depends_on:
      - wp
      - redis
    image: openbridge/nginx
    environment:
      NGINX_DOCROOT: /var/www/html
      NGINX_CONFIG: php
      NGINX_SERVER_NAME: somedomain.tld
      PHP_FPM_UPSTREAM: wp:9000
      NGINX_PROXY_UPSTREAM: wp:9000
      REDIS_UPSTREAM: redis:6379
    ports:
      - '80:80'
      - '443:443'
    volumes:
      - ./wp-web:/var/www/html
      - ./dhparam.pem:/etc/pki/tls/dhparam.pem
      - /etc/letsencrypt/live/somedomain.tld/fullchain.pem:/etc/letsencrypt/live/somedomain.tld/fullchain.pem
      - /etc/letsencrypt/live/somedomain.tld/privkey.pem:/etc/letsencrypt/live/somedomain.tld/privkey.pem
      - /etc/letsencrypt/live/somedomain.tld/chain.pem:/etc/letsencrypt/live/somedomain.tld/chain.pem
    restart: always
    networks:
      - net

  db:
    image: mysql:5.7
    volumes:
      - ./wp-db:/var/lib/mysql
    restart: always
    environment:
      MYSQL_ROOT_PASSWORD: passrandom
      MYSQL_DATABASE: wordpress
      MYSQL_USER: wordpress
      MYSQL_PASSWORD: randompass
    networks:
      - net

  wp:
    depends_on:
      - db
    image: wordpress:php7.3-fpm-alpine
    restart: always
    volumes:
      - ./wp-web:/var/www/html
    environment:
      WORDPRESS_DB_HOST: db:3306
      WORDPRESS_DB_USER: wordpress
      WORDPRESS_DB_PASSWORD: randompass
    networks:
      - net

  redis:
    image: redis:alpine
    container_name: redis
    networks:
      - net
    restart: unless-stopped
networks:
  net
volumes:
  wp-db:
`

docker ps
`CONTAINER ID        IMAGE                         COMMAND                  CREATED             STATUS              PORTS                                                  NAMES
44468c130c9b        openbridge/nginx              "/docker-entrypoint.…"   3 days ago          Up 3 days           *:80->80/tcp, *:443->443/tcp   rec_web_1
c0cb54a5774a        wordpress:php7.3-fpm-alpine   "docker-entrypoint.s…"   3 days ago          Up 3 days           9000/tcp                                               rec_wp_1
ab84677495fb        mysql:5.7                     "docker-entrypoint.s…"   3 days ago          Up 3 days           3306/tcp, 33060/tcp                                    rec_db_1
4257ef159d62        redis:alpine                  "docker-entrypoint.s…"   3 days ago          Up 3 days           6379/tcp                                               redis
`
But when i try to access somedomain.tld i get
"HTTP 502 / No backend servers found at the moment"

**Comment by tspicer** [comment]
Hi,
I noticed you set your PROXY address to your PHP service. Is there a reason you are directing NGINX to connect to PHP as a PROXY? You have `NGINX_PROXY_UPSTREAM: wp:9000`. I would normally expect a reverse proxy to be internal, so something like `NGINX_PROXY_UPSTREAM=localhost:8080`

The docs dig into this: https://github.com/openbridge/nginx#digging-into-the-env-file

Take a look at where this setting is used: https://github.com/openbridge/nginx/tree/master/conf/php/nginx/upstream.d

In a nutshell, it appears you are treating the proxy and php-fpm as the same thing. Maybe this is a typo?

I can envision certain configurations where the PROXY is a different connection, but this is an advanced setup.  Also, you may want to try using this container for PHP-FPM: https://github.com/openbridge/ob_php-fpm

It affords more flexibility and is optimized for this WP.

**Comment by tspicer** [comment]
In `nginx.conf` notice the upstream includes are distinct for each service. If PHP and PROXY are the same, I'm not sure how it would work properly.

```
upstream php-fpm      {include /etc/nginx/upstream.d/php-fpm.conf;}
upstream proxy      {include /etc/nginx/upstream.d/proxy.conf;}
upstream redis     {include /etc/nginx/upstream.d/redis.conf;}
```

**Comment by wualex** [comment]
@tspicer changed NGINX_PROXY_UPSTREAM to localhost:8080 and that made it work
i can't wrap my head around it at the moment but i'll read some more docs to understand how it really works

thank you!

**Comment by tspicer** [comment]
awesome, glad to hear.
